src/Microcks.Testcontainers/MicrocksContainerExtensions.cs

```diff
@@ -0,0 +1,162 @@
+using Microcks.Testcontainers.Model;
+using System.IO;
+using System.Net;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Text;
+using System.Text.Json;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microcks.Testcontainers;
+
+/// <summary>
+/// Extensions for MicrocksContainer.
+/// </summary>
+public static class MicrocksContainerExtensions
```

Comment:
I don't get how this class get "mix in" the MicrocksContainer one to that tests can just call `_microcksContainer.TestEndpointAsync()` and get this code invoked. Can you explain this?

Reply:
Extension methods allow you to add methods to existing types without creating a derived type, recompiling, or modifying the original type. It's super when the code doesn't belong to us.

Extension methods are static methods that are called as if they were instance methods on the extended type. For client code, there is no difference when calling an extension method and methods defined in a type.

Here I used it to reduce the content of Microckscointainer class.
The method is "public" and i use "this". 

 ```csharp
**public** static async Task<TestResult> TestEndpointAsync(**this** MicrocksContainer container, TestRequest testRequest)
 ```

However, I can refactor if you want.

Reply:
No that's fine. I was just not aware of this way of doing this. Thanks for the explanation!